src/main/java/com/atlauncher/viewmodel/impl/settings/JavaSettingsViewModel.java

```diff
@@ -0,0 +1,523 @@
+
+                        File jPath = new File(App.settings.javaPath, "bin");
+                        boolean valid = jPath.exists();
+                        javaPathCheckStateConsumer.onNext(new CheckState.Checked(valid));
+                        javaPathChanged = false;
+                        SettingsValidityManager.setValidity("javaPath", valid);
+
+                        if (!valid) {
+                            LOG.debug("javaPathCheckThread: Check thread reporting check fail");
+                        } else {
+                            SettingsManager.post();
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    private void verifyJavaParam() {
+        LOG.debug("Running javaParamCheckThread");
+        while (true) {
+            try {
+                TimeUnit.MILLISECONDS.sleep(100);
+            } catch (InterruptedException e) {
+                LOG.error("javaParamCheckThread : Failed to delay check thread", e);
+            } finally {
+                if (javaParamChanged) {
+                    javaParamCheckStateConsumer.onNext(CheckState.CheckPending);
+                    if (javaParamLastChange + javaParamCheckDelay < System.currentTimeMillis()) {
+                        // Prevent user from saving while checking
+                        setJavaParamsPending();
+                        javaParamCheckStateConsumer.onNext(CheckState.Checking);
+
+                        String params = App.settings.javaParameters;
+                        boolean valid = !(params.contains("-Xms") || params.contains("-Xmx")
+                            || params.contains("-XX:PermSize")
+                            || params.contains("-XX:MetaspaceSize"));
+                        javaParamCheckStateConsumer.onNext(new CheckState.Checked(valid));
+                        javaParamChanged = false;
+                        SettingsValidityManager.setValidity("javaParam", valid);
+
+                        if (!valid) {
+                            LOG.debug("javaParamCheckThread: Check thread reporting check fail");
+                        } else SettingsManager.post();
+                    }
+                }
+            }
+        }
+    }
+
+    @Override
+    public void onSettingsSaved() {
+        _addOnInitialRamChanged.onNext(App.settings.initialMemory);
+        _addOnMaxRamChanged.onNext(App.settings.maximumMemory);
+        _addOnPermGenChanged.onNext(App.settings.metaspace);
+        _addOnWidthChanged.onNext(App.settings.windowWidth);
+        _addOnHeightChanged.onNext(App.settings.windowHeight);
+
+        _addOnJavaPathChanged.onNext(App.settings.javaPath);
+        _addOnJavaParamsChanged.onNext(App.settings.javaParameters);
+        baseJavaInstallFolder.onNext(Optional.ofNullable(App.settings.baseJavaInstallFolder).orElse(""));
+
+        _addOnStartMinecraftMaxChanged.onNext(App.settings.maximiseMinecraft);
```

Comment:
Again why is this not `_addOnMaximiseMinecraftChanged`? A lot of these variable naming seems to be changed vs what the underlying setting is called

Reply:
I believe at an early point while rewriting the settings, I had named it such to better understand what was each setting.

Will correct soon.

Reply:
[1d9281d](https://github.com/ATLauncher/ATLauncher/pull/910/commits/1d9281dc2bc9525e29db4917a198f79d28e818a7)